From 145626141303120f5a772fe0bc809d04b8f40e1d Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Tue, 5 Jan 2021 00:40:18 -0700
Subject: [PATCH] mousepad-window: fix a crash where...

... if there is only one document, the search function will crash the
program
---
 mousepad/mousepad-window.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mousepad/mousepad-window.c b/mousepad/mousepad-window.c
index 7be0fa4..078e2bc 100644
--- a/mousepad/mousepad-window.c
+++ b/mousepad/mousepad-window.c
@@ -2389,6 +2389,12 @@ mousepad_window_notebook_removed (GtkNotebook     *notebook,
   mousepad_disconnect_by_func (document->textview, mousepad_window_menu_textview_popup, window);
   mousepad_disconnect_by_func (document->textview, mousepad_window_enable_edit_actions, window);
 
+  /* reset the reference to NULL to avoid illegal memory access */
+  if (window->previous == document)
+    {
+      window->previous = NULL;
+    }
+
   /* window contains no tabs: save geometry and destroy it */
   if (gtk_notebook_get_n_pages (notebook) == 0)
     {
-- 
2.30.0

